WordPress/gutenberg
# ColorListPicker: refactor to TypeScript

## What?
Refactor `ColorListPicker` component to TypeScript

Part of https://github.com/WordPress/gutenberg/issues/35744

## Why?

The refactor to TypeScript has many benefits (auto-generated docs, static linting and error prevention, better IDE experience). See https://github.com/WordPress/gutenberg/issues/35744 for more details

## How?
Followed the steps in the [TypeScript migration guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md?rgh-link-date=2022-11-29T14%3A45%3A34Z#refactoring-a-component-to-typescript)

## Testing Instructions
- Ensure all tests still pass
- Confirm Storybook examples and editor examples work as expected. For now, it can be found/tested in components like the Duotone filter.

### Testing Instructions for Keyboard

TBD
